test/integration/infrastructure/infrastructure_test.go

```diff
@@ -82,193 +80,519 @@ func validateFlags() {
+			err = c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: "infrastructure"}, infra)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(infra.Status.LastError).NotTo(BeNil())
+			Expect(infra.Status.LastError.Description).To(ContainSubstring("invalid VPC attributes: `enableDnsHostnames` must be set to `true`"))
+		})
+
+		It("should successfully create and delete", func() {
+			enableDnsHostnames := true
+			vpcID, err := prepareNewVPC(ctx, logger, awsClient, vpcCIDR, enableDnsHostnames)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(vpcID).NotTo(BeEmpty())
+
+			providerConfig, err := newProviderConfig(awsv1alpha1.VPC{
+				ID:               &vpcID,
+				GatewayEndpoints: []string{s3GatewayEndpoint},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			AddCleanupAction(func() {
+				err := teardownVPC(ctx, logger, awsClient, vpcID)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			namespace, err := generateNamespaceName()
+			Expect(err).NotTo(HaveOccurred())
+
+			err = runTest(ctx, logger, c, namespace, providerConfig, decoder, awsClient)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+})
+
+func runTest(ctx context.Context, logger *logrus.Entry, c client.Client, namespaceName string, providerConfig *awsv1alpha1.InfrastructureConfig, decoder runtime.Decoder, awsClient *awsclient.Client) error {
+	var (
+		namespace                 *corev1.Namespace
+		cluster                   *extensionsv1alpha1.Cluster
+		infra                     *extensionsv1alpha1.Infrastructure
+		infrastructureIdentifiers infrastructureIdentifiers
+	)
+
+	AddCleanupAction(func() {
 		By("delete infrastructure")
-		Expect(infrastructure.Delete(ctx, logger, restConfig, c, infra)).NotTo(HaveOccurred())
-		Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).NotTo(HaveOccurred())
+		Expect(client.IgnoreNotFound(c.Delete(ctx, infra))).To(Succeed())
+
+		By("wait until infrastructure is deleted")
+		err := common.WaitUntilExtensionCRDeleted(
+			ctx,
+			c,
+			logger,
+			func() extensionsv1alpha1.Object { return &extensionsv1alpha1.Infrastructure{} },
+			"Infrastructure",
+			infra.Namespace,
+			infra.Name,
+			10*time.Second,
+			16*time.Minute,
+		)
+		Expect(err).NotTo(HaveOccurred())
 
 		By("verify infrastructure deletion")
 		verifyDeletion(ctx, awsClient, infrastructureIdentifiers)
 
-		aws.InternalChartsPath = internalChartsPath
+		Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+		Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
 	})
 
-	BeforeEach(func() {
-		scheme = runtime.NewScheme()
-		Expect(awsinternal.AddToScheme(scheme)).To(Succeed())
-		Expect(awsv1alpha1.AddToScheme(scheme)).To(Succeed())
-		decoder = serializer.NewCodecFactory(scheme).UniversalDecoder()
-	})
+	By("create namespace for test execution")
+	namespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if err := c.Create(ctx, namespace); err != nil {
+		return err
+	}
```

Comment:
```suggestion
	Expect(c.Create(ctx, namespace)).To(Succeed())
```

Reply:
Wrt to the end result, there is no change as the `runTest` func returns err and it is being checked in the test itself. I had to adapt the `runTest` func to be able to return err as we now have a test case that expects err to have occurred. And I didn't want to mix 
```
Expect(err).NotTo(HaveOccurred)
```

statements with err propagation

```
if err != nil {
    return err
} 
```

in the same func and that's why I decided to be consistent in the func and use only error propagation.

Let me know if that makes sense to you.

Reply:
Yes, sure, it's fine with me. My point was also only consistency because I saw https://github.com/gardener/gardener-extension-provider-aws/pull/157/files#diff-10becb4271192c12b5fa414024464971R273-R274, but those are actually statements for the `CleanupAction` which I didn't see earlier.

Fine with me, all good. :)